### ISSUE TITLE
Improve PanacheQuery mocking

### ISSUE THREAD

**Issue by loicmathieu** [quarkusio/quarkus#13251]
**Description**
Reading our current documentation, it is not evident how to mock a PanacheQuery object.

On the following SO [question](https://stackoverflow.com/questions/64765163/how-to-mock-panachequeryentity-using-quarkus-with-hibernate-omr-panachereposit/64803160#64803160) I proposes the following solution, that is a little verbose: 

```
PanacheQuery query = Mockito.mock(PanacheQuery.class);
Mockito.when(query.page(Mockito.any()).thenReturn(query);
Mockito.when(query.list()).thenReturn(labels);
```

**Implementation ideas**
Document the propose solution inside the Mock section of all Panache guide.

Or provides an easy way to mock a PanacheQuery (via a list backed PanacheQuery implementation for example, but as there a different PanacheQuery for MongoDB, Hibenrate, Kotlin, reactive, ... this canbe hard to implement).

**Comment by GiuseppeScaramuzzino** [comment]
I agree with @loicmathieu to introduce an easy way to mock the PanacheQuery implementation or directly a Mock for PanacheRepository.

In my case I need to test the .find() of my PanacheRepository<Label> in this way (thanks to @loicmathieu ):

```
PanacheQuery query = Mockito.mock(PanacheQuery.class); 
Mockito.when(query.page(Mockito.any())).thenReturn(query);
Mockito.when(query.firstResultOptional()).thenReturn(Optional.of(label));
when(labelRepository.find("name", "LABEL#01")).thenReturn(query);
```

Link to the StackOverlow issue: https://stackoverflow.com/questions/64765163/how-to-mock-panachequeryentity-using-quarkus-with-hibernate-omr-panachereposit?answertab=active#tab-top

**Comment by FroMage** [comment]
I still think it's better to use an in-memory DB rather than mock the entities, but if you really must, then I agree that we could add `public static <T> PanacheQuery fromResults(List<T> results)` to every variant of `PanacheQuery` to make this easier.

**Comment by SirSkizo** [comment]
Instead of using `Mockito.mock()`, you might use `@InjectMock`

```java
  @InjectMock
  private TRepository tRepository;

  @BeforeAll
  public void setup() {
    Mockito.when(tRepository.tMethod(anyString(), anyString())).thenReturn(
        <anything>
    );
  }
```

**Comment by Jaland** [comment]
Just adding my voice I think it would be really nice if there was just a simple way to mock a `.find()`.

And regardless of the folks who think an in-memory db is the better option there are lots of reasons that may not be possible or easily implementable for a unit test, so having a way I can mock exactly what I want the output of a query to be should be a pretty basic feature.

**Comment by FroMage** [comment]
So, what do you think of this proposed solution?

```java
Mockito.when(labelRepository.find("name", "LABEL#01")).thenReturn(PanacheQuery.fromResults(label));
```

**Comment by RayanArgolo03** [comment]
Other idea: mock project() method to use in Pagination later

**Comment by FroMage** [comment]
I'm not sure I understand your suggestion, could you explain?
